specs/core/0_beacon-chain.md

```diff
@@ -1596,41 +1611,35 @@ _Note_: If there are skipped slots between a block and its parent block, run the
 
 ### Per-slot processing
 
-Below are the processing steps that happen at every slot.
-
-#### Slot
+Below are the processing steps that happen at every `slot > GENESIS_SLOT`.
 
+* Set `state.latest_state_roots[state.slot % SLOTS_PER_BATCHING] = hash(state)`.
+* Set `state.latest_block_roots[state.slot % SLOTS_PER_BATCHING] = get_block_root(state, state.slot)`.
```

Comment:
The call to `get_block_root` will fail here due to the assertion -- `assert slot < state.slot <= slot + SLOTS_PER_BATCHING`. This is an unfortunate consequence of the validity conditions of some helpers are dependent upon _when_ in the state transition function they are called.

Reply:
Also, at best, this just overwrites a value with itself. What is the intention here?

Reply:
Seems like maybe we should be doing the following after the `state.slot += 1`

```
* Set `state.latest_block_roots[state.slot % SLOTS_PER_BATCHING] = get_block_root(state, state.slot - 1)`.
```

Reply:
I changed things around. Does that make more sense?